### PR TITLE
add search functionality to project index page

### PR DESCRIPTION
**Pull request by fdiblen**
# Pull request details

## List of related issues or pull requests

Refs: #663 

## Describe the changes made in this pull request
- Added search bar to project index template
- Added search functions
- Added search section style

<details open>
<summary>Screenshot</summary>
<br>

rsd_project_search_

</details>

## Instructions to review the pull request

1. Checkout the branch and start the stack
```
git checkout 663-search-bar
docker-compose down --volumes --remove-orphan
docker-compose build frontend
docker-compose up -d
docker-compose logs --follow
``
1. Compare the result with the screenshot
1. Go to [https://localhost/projects/](https://localhost/projects/) and search a project using title or subtitle

### CODE REVIEW

File: frontend/static/scripts/project_index_scripts.js [hunk 1]
```diff
@@ -25,11 +25,19 @@ function initOverview(projectsData, statusChoicesData) {
...
+            return fields.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1;
+        }
+    }
+
     var v = new Vue({
         el: '#project_index_page',
         delimiters: ["[[", "]]"],
         methods: {
-            log: console.log,
+            log: console.log,
```
Comment (sverhoeven) [inline frontend/static/scripts/project_index_scripts.js:40]: Dont add empty space at end of line
```suggestion
            log: console.log,
```

File: frontend/templates/project_index/template.html [hunk 2]
```diff
@@ -50,18 +68,13 @@ <h1 class="title">Projects</h1>
...
-                  tag="ul"
-                  class="filter-list"
-                  v-bind:css="false"
-                  v-on:before-enter="filtersBeforeEnter"
-                  v-on:enter="filtersEnter"
-                  v-on:before-leave="filtersBeforeLeave"
-                  v-on:leave="filtersLeave"
-                >
-                  <li v-for="(status, index) in statusesWithCount" v-bind:data-index="index" :key="status" v-if="statusesFilterOpen" :class="status.count === 0 ? 'empty' : ''">
+                <transition-group tag="ul" class="filter-list" v-bind:css="false" v-on:before-enter="filtersBeforeEnter"
```
Comment (sverhoeven) [inline frontend/templates/project_index/template.html:71]: Seems your ran some formatter, this makes it hard to see what is different. The search bar addition should not have changed anything below line 40.

**Review by sverhoeven — APPROVED** [review]
Search works for me, even on phone.

Formatter made changes more difficult, but changes are reasonable.

File: frontend/templates/project_index/template.html (hunk 2 above)
Comment (fdiblen) [reply frontend/templates/project_index/template.html:71]: Yes, there are not any changes. Only formatting.

**Pull request merged**